src/__test__/App.test.js

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import App from "../App";
+import userEvent from "@testing-library/user-event";
+
+test("should perform a basic dice calculation", async () => {
+  const { getByText, getByLabelText } = render(<App />);
```

Comment:
@martypowell Would this be a point where we can pass in a stub class that provides the `calculate(formula)` method, so that your UI tests aren't at the mercy of the main Formula engine?  That way you can assert specifically that the formula is sent, and that the response is added to history.